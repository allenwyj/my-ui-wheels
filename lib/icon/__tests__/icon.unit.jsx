import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Icon from '../icon.component';

describe('icon tests', () => {
  it('expects a svg', () => {
    const json = renderer.create(<Icon name="alipay" />).toJSON();
    expect(json).toMatchSnapshot();
  });

  it('onClick', () => {
    const fn = jest.fn();
    const component = mount(<Icon name="alipay" onClick={fn} />);
    component.find('svg').simulate('click');
    expect(fn).toBeCalled();
  });
});
