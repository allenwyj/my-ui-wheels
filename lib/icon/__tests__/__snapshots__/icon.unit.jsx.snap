// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon tests expects a svg 1`] = `
<svg
  className="sui-icon"
>
  <use
    xlinkHref="#i-alipay"
  />
</svg>
`;
